Introduce #readFromContext: for all variables

dionisiydk opened the pull request:
- introduces #readInContext: for all variables and refactors #lookupSymbol:.
As result it fixes #lookupSymbol: to work with all kind of slots (it was only working for simple inst vars).
- introduce write:inContext for all variables.

Having pair of those methods we can get read and write any kind of variable having context instance.

dionisiydk committed:
Introduce #readFromContext: for all variables and refactor #lookupSymbol using it.
It removes one user of #isGlobal test.

dionisiydk commented on src/Debugging-Core/Context.extension.st:
```diff
@@ -22,15 +22,13 @@ Context >> callPrimitive: primNumber [
 { #category : #'*Debugging-Core' }
 Context >> lookupSymbol: aSymbol [
```
I think we should rename this method to something like #valueOfSymbol:. I can do it in another PR. Suggestions for name are welcome.

MarcusDenker replied:
yes, the name is not good... maybe valueForVariableNamed? valueForSymbol:?

dionisiydk replied:
and we need another name to modify the value. Basically they will be generalized version of tempNamed:/tempNamed:put:

dionisiydk replied:
I think #read/#write are good words to use there:
- aContext readVarNamed: aSymbol
- aContext write: aValue toVarNamed: aSymbol

dionisiydk committed:
introduce #write:toContext for all variables and refactor tempNamed:put:
Now we have a pair of methods #readFromContext: and write:toContext: to modify variables having a context instance.

dionisiydk committed:
renames: readInContext: and write:inContext:

MarcusDenker reviewed (approved):
Good! Sadly the CI is down...

MarcusDenker commented:
testReifyAssignmentTempOperationBefore is failing

MarcusDenker commented:
the reason is that OCTempVariable #write:inContext:  misses a return at the end.

(writing a variable used to return the value.. no idea if that is bad or good)

dionisiydk committed:
Fix test: return asignedValue explicitly from RFTempWrite. 
And remove return of #asignedValue from all #write  methods to not confuse users
